src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +230,23 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param types      The types of logs being requested. This should correspond to values in the enum MTRDiagnosticLogType.
+ * @param timeout    The timeout in minutes for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ *                   If the timeout is nil, it will never expire and completion will not be called until
+ *                   the logs are fully retrieved.
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the file paths for the requested logs.
+ */
+- (void)getDiagnosticLogsOfTypes:(NSArray<NSNumber *> *)types
```

Comment:
```suggestion
- (void)downloadDiagnosticLogsOfTypes:(NSArray<NSNumber *> *)types
```

Reply:
Or maybe request? I'm just not a fan of 'get' ;)

Reply:
For others reference too, these are good examples to reference: 

https://developer.apple.com/documentation/foundation/nsurlsessiondownloaddelegate/1411575-urlsession?language=objc
https://developer.apple.com/documentation/foundation/nsurlsession/1411481-downloadtaskwithrequest?language=objc

Reply:
"fetch" or "download" seem better than "get" to me